Broken tests: `set_readable_signals` does not accept `Motor`

Running the tests in `dodal` main against `ophy_async` main gives the following error:
```
  File "/scratch/ffv81422/hyperion/dodal/src/dodal/beamlines/i22.py", line 119, in slits_6
    return numbered_slits(
           ^^^^^^^^^^^^^^^
  File "/scratch/ffv81422/hyperion/dodal/src/dodal/utils.py", line 101, in wrapper
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/scratch/ffv81422/hyperion/dodal/src/dodal/beamlines/_device_helpers.py", line 27, in numbered_slits
    return device_instantiation(
           ^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/ffv81422/hyperion/dodal/src/dodal/utils.py", line 101, in wrapper
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/scratch/ffv81422/hyperion/dodal/src/dodal/beamlines/beamline_utils.py", line 102, in device_instantiation
    device_instance = device_factory(
                      ^^^^^^^^^^^^^^^
  File "/scratch/ffv81422/hyperion/dodal/src/dodal/devices/slits.py", line 18, in __init__
    self.set_readable_signals(
  File "/scratch/ffv81422/hyperion/dodal/.venv/lib/python3.11/site-packages/ophyd_async/core/standard_readable.py", line 68, in set_readable_signals
    self.add_readables(read, wrapper=HintedSignal)
  File "/scratch/ffv81422/hyperion/dodal/.venv/lib/python3.11/site-packages/ophyd_async/core/standard_readable.py", line 209, in add_readables
    obj = wrapper(readable)
          ^^^^^^^^^^^^^^^^^
  File "/scratch/ffv81422/hyperion/dodal/.venv/lib/python3.11/site-packages/ophyd_async/core/standard_readable.py", line 238, in __init__
    assert isinstance(signal, SignalR), f"Expected signal, got {signal}"
AssertionError: Expected signal, got <ophyd_async.epics.motion.motor.Motor object at 0x7f423024a090>
```

This is happening because `Slits` is passing a `Motor` into `set_readable_signals`, which only accepts a `Signal`. @coretl do we expect `Motor` to be able to be passed in? In which case is this an `ophyd-async` error?

## Acceptance Criteria
* Tests are passing on main

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,7 +13,7 @@ classifiers = [
 description = "Ophyd devices and other utils that could be used across DLS beamlines"
 dependencies = [
     "ophyd",
-    "ophyd-async>=0.3a2",
+    "ophyd-async>=0.3a3",
     "bluesky",
     "pyepics",
     "dataclasses-json",

--- a/src/dodal/devices/slits.py
+++ b/src/dodal/devices/slits.py
@@ -9,18 +9,9 @@ class Slits(StandardReadable):
     """
 
     def __init__(self, prefix: str, name: str = "") -> None:
-        self.x_gap = Motor(prefix + "X:SIZE")
-        self.y_gap = Motor(prefix + "Y:SIZE")
-        self.x_centre = Motor(prefix + "X:CENTRE")
-        self.y_centre = Motor(prefix + "Y:CENTRE")
-
-        # Type ignore pending: https://github.com/bluesky/ophyd-async/issues/192
-        self.set_readable_signals(
-            read=[
-                self.x_gap,
-                self.y_gap,
-                self.x_centre,
-                self.y_centre,
-            ]  # type: ignore
-        )
+        with self.add_children_as_readables():
+            self.x_gap = Motor(prefix + "X:SIZE")
+            self.y_gap = Motor(prefix + "Y:SIZE")
+            self.x_centre = Motor(prefix + "X:CENTRE")
+            self.y_centre = Motor(prefix + "Y:CENTRE")
         super().__init__(name)

